File: apps/java/src/main/java/chapter02/TrainTestSplit.java
package chapter02;

import java.util.List;

/** 訓練データとテストデータ。X は特徴量の型、T は正解ラベルの型。 */
public record TrainTestSplit<X, T>(List<X> xTrain, List<X> xTest, List<T> tTrain, List<T> tTest) {
  public TrainTestSplit {
    xTrain = List.copyOf(xTrain);
    xTest = List.copyOf(xTest);
    tTrain = List.copyOf(tTrain);
    tTest = List.copyOf(tTest);
  }
}

File: apps/csharp/src/MachineLearning/Chapter02/TrainTestSplit.cs
namespace MachineLearning.Chapter02;

/// <summary>訓練データとテストデータ。TX は特徴量の型、TT は正解ラベルの型。</summary>
public sealed record TrainTestSplit<TX, TT>(
    IReadOnlyList<TX> XTrain,
    IReadOnlyList<TX> XTest,
    IReadOnlyList<TT> TTrain,
    IReadOnlyList<TT> TTest);
